hugr/src/algorithm/const_fold.rs

```diff
@@ -78,26 +78,27 @@ impl ConstFoldConfig {
         .map_err(|err| ConstFoldError::verify_err(label, err))
     }
 
-    /// TODO
+    /// Run the Constant Folding pass.
     pub fn run(&self, h: &mut impl HugrMut, reg: &ExtensionRegistry) -> Result<(), ConstFoldError> {
         self.verify_impl("input", h, reg)?;
         loop {
-            // would be preferable if the candidates were updated to be just the
-            // neighbouring nodes of those added.
-            let rewrites = find_consts(h, h.nodes(), reg).collect_vec();
-            if rewrites.is_empty() {
+            // We can only safely apply a single replacement. Applying a
```

Comment:
yes, the original code was based on the assumption that `find_consts` would only return rewrites for ops that _only_ had const inputs (which would I think be safe to apply all of them?)  but that's not what find_consts does